Tryouts/Plugins/ApplicationPlugins/process explorer/backend/SuperRPC-POC/MiddlewareV2.cs

```diff
@@ -0,0 +1,201 @@
+﻿// Morgan Stanley makes this available to you under the Apache License,
+// Version 2.0 (the "License"). You may obtain a copy of the License at
+// 
+//      http://www.apache.org/licenses/LICENSE-2.0.
+// 
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership. Unless required by applicable law or agreed
+// to in writing, software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions
+// and limitations under the License.
+
+using System.Net;
+using System.Net.WebSockets;
+using LocalCollector.Communicator;
+using ProcessExplorer.Factories;
+using Super.RPC;
+using SuperRPC_POC.ClientBehavior;
+using SuperRPC_POC.Infrastructure;
+
+namespace SuperRPC_POC;
+
+public class SuperRpcWebSocketMiddlewareV2
+{
+    private readonly RequestDelegate _next;
+    private readonly IModuleLoaderInformationReceiver _collector;
+
+    public SuperRpcWebSocketMiddlewareV2(RequestDelegate next, IModuleLoaderInformationReceiver collector) 
+    {
+        _next = next;
+        _collector = collector;
+
+        //fire - and - forget
+        _collector.SubscribeToProcessExplorerTopicAsync();
+        _collector.SubscribeToProcessExplorerChangedElementTopicAsync();
+        _collector.SubscribeToSubsystemsTopicAsync();
+        _collector.SubscribeToRuntimeInformationTopicAsync();
+    }
+
+    private SuperRPC SetupRpc(RPCReceiveChannel channel, ICommunicator? communicator)
+    {
+        var rpc = new SuperRPC(() => Guid.NewGuid().ToString("N"));
+        SuperRPCWebSocket.RegisterCustomDeserializer(rpc);
+        rpc.Connect(channel);
+
+        if (communicator != null)
+        {
+            rpc.RegisterHostObject("communicator", communicator, new ObjectDescriptor
+            {
+                Functions = new FunctionDescriptor[] { "AddRuntimeInfo",
```

Comment:
Neatpick: make the arangment of the members consistent. AddRuntimeInfo should be in a new line, as well as the } in line 57.